### PR TITLE
fix(core): prevent tool-search rollout poisoning

### PR DESCRIPTION
**Pull request by apanasenko-oai**
## Why

Malformed client-executed `tool_search_call.arguments` can be returned by the Responses service and persisted verbatim. Once written to a rollout, the malformed call is replayed on every follow-up and cold resume, leaving the session permanently unusable. This prevents future occurrences of [#26379](https://github.com/openai/codex/issues/26379); already-poisoned rollouts remain out of scope.

## What Changed

- Keep `rawResponseItem/completed` faithful to the original server item while canonicalizing client tool-search calls before durable history and rollout persistence.
- Re-serialize valid arguments through `SearchToolCallParams`, replace malformed arguments with a bounded placeholder, and keep missing or empty call IDs out of durable history.
- Produce a matching empty `tool_search_output` for malformed calls with usable IDs, and keep parse diagnostics payload-free.
- Leave server-executed tool-search calls unchanged.
- Add an end-to-end regression test covering malformed arguments, unknown fields, missing and empty call IDs, server execution, raw event fidelity, follow-up pairing, rollout contents, and cold resume.

## Verification

`just test -p codex-core --retries 0 malformed_tool_search_is_repaired_before_persistence_and_resume` passes and exercises an oversized multilingual argument key through persistence and cold resume.

### CODE REVIEW

**Review by montazeri-oai — APPROVED** [review]

File: codex-rs/core/src/session/mod.rs [hunk 1]
```diff
@@ -2777,18 +2776,22 @@ impl Session {
         turn_context: &TurnContext,
         items: &[ResponseItem],
     ) {
-        let items = self.prepare_conversation_items_for_history(turn_context, items);
-        let items = items.as_ref();
+        let raw_items = self.prepare_raw_response_items(turn_context, items);
+        let history_items =
+            tool_search_history::prepare_for_durable_history(&self.thread_id, raw_items.as_ref());
```
[the remark this answers is not in the document]
Comment (apanasenko-oai) [reply codex-rs/core/src/session/mod.rs]: [from Codex]: No code change is needed for this thread. PR #30618 intentionally prevents future poisoning and does not migrate already-poisoned rollouts, as stated in the PR body and approved implementation plan. Commit 22aadb61fe canonicalizes newly recorded calls before persistence, and its regression test cold-resumes a rollout created after the fix. Repairing legacy poisoned rollouts remains separate work.
